com.unity.ml-agents.extensions/Runtime/Input/InputActuatorComponent.cs

```diff
@@ -0,0 +1,376 @@
+                    }
+                }
+            }
+
+            var controlSchemeName = mlAgentsControlSchemeName + localId;
+            var inputControlScheme = new InputControlScheme(
+                controlSchemeName,
+                deviceRequirements);
+
+            return inputControlScheme;
+        }
+
+#pragma warning disable 672
+        /// <inheritdoc cref="ActuatorComponent.CreateActuator"/>
+        public override IActuator CreateActuator() { return null; }
+#pragma warning restore 672
+
+        /// <inheritdoc cref="IActuator.ActionSpec"/>
+        public override ActionSpec ActionSpec => m_ActionSpec;
+
+        /// <summary>
+        /// Creates a virtual <see cref="InputDevice"/> with a layout that is based on the <see cref="InputAction"/>s
+        /// in the <see cref="InputActionMap"/> specified by <see cref="PlayerInput.defaultActionMap"/>.
+        /// </summary>
+        /// <param name="layoutName">The custom layout name to use for the <see cref="InputDevice"/> to create.</param>
+        /// <param name="interfaceName">The made up interface name we are using for these virtual devices.  Right now
+        /// it is equal to <see cref="mlAgentsDeviceName"/> + <see cref="BehaviorParameters.BehaviorName"/></param>
+        /// <param name="localId"></param>
+        /// <returns></returns>
+        internal static InputDevice CreateDevice(string layoutName, string interfaceName, uint localId)
+        {
+            // See if our device layout was already registered.
+            var existingLayout = InputSystem.TryFindMatchingLayout(new InputDeviceDescription
+            {
+                interfaceName = interfaceName
+            });
+
+            // Load the device layout based on the controls we created previously.
+            if (string.IsNullOrEmpty(existingLayout))
+            {
+                InputSystem.RegisterLayoutMatcher(layoutName, new InputDeviceMatcher()
+                    .WithInterface($"^({interfaceName}[0-9]*)"));
+            }
+
+            // Actually create the device instance.
+            var device = InputSystem.AddDevice(
+                new InputDeviceDescription
+                {
+                    interfaceName = interfaceName,
+                    serial = $"{localId}"
+                }
+            );
+            return device;
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="defaultMap"></param>
+        /// <param name="layoutName"></param>
+        /// <param name="localId"></param>
+        /// <param name="behaviorParameters"></param>
+        /// <returns></returns>
+        internal static IActuator[] GenerateActionActuatorsFromAsset(
+            InputActionMap defaultMap,
+            string layoutName,
+            uint localId,
+            BehaviorParameters behaviorParameters)
+        {
+            // TODO does this need to change based on the action map we use?
```

Comment:
Probably want to build a device mapping all actions in the asset. Simple to change, though. Controls can simply use "map_action" as their naming format.

Reply:
I want to reduce the number of actions I map to as not to overload the neural network with actions it doesn't need to know